pulp_rpm/tests/functional/api/test_publish.py

```diff
@@ -333,3 +342,141 @@ def _get_updateinfo_xml_path(root_elem):
         ]
         xpath = '{{{}}}location'.format(RPM_NAMESPACES['metadata/repo'])
         return data_elems[0].find(xpath).get('href')
+
+
+class ChecksumTypeTestCase(unittest.TestCase):
+    """Publish repository and validate the updateinfo.
+
+    This Test does the following:
+
+    1. Create a rpm repo and a remote.
+    2. Sync the repo with the remote.
+    3. Publish and distribute the repo.
+    4. Verify the checksum types
+
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables."""
+        delete_orphans()
+        cls.cfg = config.get_config()
+        cls.client = gen_rpm_client()
+        cls.repo_api = RepositoriesRpmApi(cls.client)
+        cls.remote_api = RemotesRpmApi(cls.client)
+        cls.publications = PublicationsRpmApi(cls.client)
+        cls.distributions = DistributionsRpmApi(cls.client)
+
+    def get_checksum_types(self, **kwargs):
+        """Sync and publish an RPM repository."""
+        package_checksum_type = kwargs.get("package_checksum_type")
+        metadata_checksum_type = kwargs.get("metadata_checksum_type")
+        policy = kwargs.get("policy", "immediate")
```

Comment:
it would be great to test on demand policy as well

Reply:
I tried to test it, but I had some problem with delete_orphans and the randomness of the test, if the on_demand starts first it passes, if not, it fails

Reply:
yes, we whether need to call orphan clean up between the tests, or make sure the first test on demand and then immediate, because after immediate our artifacts are populated.